py/desitarget/targets.py

```diff
@@ -520,11 +520,22 @@ def calc_priority(targets, zcat, obscon):
 
         # ADM Secondary targets.
         if scnd_target in targets.dtype.names:
+            # APC Secondary target bits only drive updates to targets with specific DESI_TARGET bits
+            # APC See https://github.com/desihub/desitarget/pull/530
+            scnd_update = (targets[desi_target] & desi_mask['SCND_ANY']) != 0
```

Comment:
Line 525 not strictly necessary given that individual scnd_mask names are checked below, but seems harmless way to create `scnd_update` vs. `np.repeat(False,len(targets))`.